File: web/src/api/siteVariable.ts
import apiClient from './client';
import type { 
  SiteVariable,
  CreateSiteVariableRequest,
  UpdateSiteVariableRequest,
  GetSiteVariableListParams,
  SiteVariableListResponse,
  GetSiteVariableByKeyResponse
} from '@/types/siteVariable';
import type { ApiResponse } from '@/types/global';

export const siteVariableAPI = {
  // ========== 管理员接口 ==========
  
  // 创建网站变量（管理员）
  createSiteVariable: (data: CreateSiteVariableRequest): Promise<ApiResponse<null>> => {
    return apiClient.post('/api/admin/site-variables', data);
  },

  // 更新网站变量（管理员）
  updateSiteVariable: (id: number, data: UpdateSiteVariableRequest): Promise<ApiResponse<null>> => {
    return apiClient.put(`/api/admin/site-variables/${id}`, data);
  },

  // 删除网站变量（管理员）
  deleteSiteVariable: (id: number): Promise<ApiResponse<null>> => {
    return apiClient.delete(`/api/admin/site-variables/${id}`);
  },

  // 获取网站变量列表（管理员）
  getSiteVariableList: (params?: GetSiteVariableListParams): Promise<ApiResponse<SiteVariableListResponse>> => {
    return apiClient.get('/api/admin/site-variables', { params });
  },

  // 通过ID获取网站变量详情（管理员）
  getSiteVariableByID: (id: number): Promise<ApiResponse<SiteVariable>> => {
    return apiClient.get(`/api/admin/site-variables/${id}`);
  },

  // ========== 公开接口 ==========
  
  // 通过key获取网站变量（公开）
  getSiteVariableByKey: (key: string): Promise<ApiResponse<GetSiteVariableByKeyResponse>> => {
    return apiClient.get('/api/public/site-variables/by-key', { params: { key } });
  },
};

File: server/router/sitevariable.go
package router

import (
	"server/api/sitevariable"

	"github.com/gin-gonic/gin"
)

// InitSiteVariableRouter 初始化网站变量相关路由
func InitSiteVariableRouter(privateGroup *gin.RouterGroup, publicGroup *gin.RouterGroup, adminGroup *gin.RouterGroup) {
	// 管理员路由 - 网站变量的增删查改
	AdminSiteVariableRouter := adminGroup.Group("/api/admin/site-variables")
	{
		AdminSiteVariableRouter.POST("", sitevariable.CreateSiteVariable)       // 创建网站变量
		AdminSiteVariableRouter.PUT("/:id", sitevariable.UpdateSiteVariable)    // 更新网站变量
		AdminSiteVariableRouter.DELETE("/:id", sitevariable.DeleteSiteVariable) // 删除网站变量
		AdminSiteVariableRouter.GET("", sitevariable.GetSiteVariableList)       // 获取网站变量列表
		AdminSiteVariableRouter.GET("/:id", sitevariable.GetSiteVariableByID)   // 通过ID获取网站变量详情
	}

	// 公开路由 - 非管理员通过key查询
	PublicSiteVariableRouter := publicGroup.Group("/api/public/site-variables")
	{
		PublicSiteVariableRouter.GET("/by-key", sitevariable.GetSiteVariableByKey) // 通过key获取网站变量
	}
}
